packages/core/src/react-integration/hooks/useResource.ts

```diff
@@ -29,14 +30,29 @@ function useOneResource<
   DenormalizeNullable<Shape['schema']>,
   Denormalize<Shape['schema']>
 > {
-  const maybePromise = useRetrieve(fetchShape, params);
   const state = useContext(StateContext);
-  const [denormalized, ready] = useDenormalized(fetchShape, params, state);
+  const [denormalized, ready, notDeleted] = useDenormalized(
+    fetchShape,
+    params,
+    state,
+  );
   const error = useError(fetchShape, params, ready);
 
-  if (!hasUsableData(ready, fetchShape) && maybePromise) throw maybePromise;
+  const maybePromise = useRetrieve(fetchShape, params, !notDeleted && !error);
+
   if (error) throw error;
 
+  if (
+    !hasUsableData(
+      fetchShape,
+      ready,
+      notDeleted,
+      useMeta(fetchShape, params)?.invalidated,
+    ) &&
+    maybePromise
+  )
+    throw maybePromise;
```

Comment:
nit: Personally I'm a big fan of eslint/ prettier enforcing braces if the condition is multi-line

Reply:
mm, excellent point. whcih rule is that, i'll upstream to anansi's root config.